docs/DeveloperGuide.md

```diff
@@ -462,7 +462,76 @@ InternHunter only lets users create applications for internships already added t
     * Cons:
         * High risk of data inconsistency due to the linkage between company and application lists.
 
-### Match Command
+### Match Command feature
+
+#### What it is
+Users are able to execute a command to generate a list of matching internships that matches their current profile
+skills. This matching is done by accumulating the list of profile items that has the category `SKILL` and
+filtering the list of internships from them. Remaining internships are those that consist of at least one
+`Requirement` that matches the user's list of skills. 
+
+**Command format:** `match`
+
+#### Implementation
+
+`MatchCommand` is a class that extends the `Command` _abstract_ class. It has a dependency to the `Model`
+interface as it relies on getting the profile list and company item list from the model.
+
+Here is a class diagram to show how the `MatchCommand` is implemented:
+
+![MatchCommandClassDiagram](images/MatchCommandClassDiagram.png)
+
+This is how the `MatchCommand#execute()` method works upon execution:
+
+1. The list of profile items and company items are first obtained via the `Model#getProfileItemList()` method and
+`Model#getCompanyItemList()` respectively.
+2. Then, the list of profile skills that the user has is obtained via a self-invocation to
+`MatchCommand`'s own `getSkillList(profileItemList)` method.
+3. Next, the list of all internships from the list of companies is obtained via its own
+`getInternshipList(companyItemList)` method.
+4. Then, the `MatchCommand`'s own `getMatchingInternships(fullInternshipList, currentSkillList)` is invoked to obtain
```

Comment:
i think you have changed `currentSkillList` to `skillList` in your implementation?

Reply:
yup changed